src/ws/update-motd

```diff
@@ -12,13 +12,25 @@ port=${port:-9090}
 # hostname from cmdline, then `hostname -f`
 hostname=${2:-$(hostname -f)}
 
+# determine the first interface with a default route
+interface6=${3:-$(ip -o -6 route get :: 2>/dev/null | sed -e 's/.*dev \([^ ]*\) .*/\1/')}
+# use the interface to determine the global ip6 address
+if [ ! -z $interface6 ]; then
+    ip6=${3:-$(ip -o -6 addr show dev $interface6 scope global | sed -e 's/.*inet6 \([^ ]*\) .*/\1/' | sed -e 's/\/.*//')}
+fi
+
```

Comment:
This seems unnecessarily complicated.  Why not just adapt the ip4 line to something like:

    ip route get 2001:: | sed -e 's/.*src \([^ ]*\) .*/\1/

That works for me.